src/Command/GenerateJsonCommand.php

```diff
@@ -119,13 +119,14 @@ private function generatePrestaShopModulesJson(
             }
         }
 
+        $modulesPath = $this->jsonDir . '/modules/';
+        if (!is_dir($modulesPath)) {
+            mkdir($modulesPath);
+        }
+
         foreach ($infos as $prestashopVersion => $modules) {
             $output->writeln(sprintf('<info>Generate json for PrestaShop %s</info>', $prestashopVersion));
-            $prestashopVersionPath = $this->jsonDir . '/' . $prestashopVersion;
-            if (!is_dir($prestashopVersionPath)) {
-                mkdir($prestashopVersionPath);
-            }
-            file_put_contents($prestashopVersionPath . '/modules.json', json_encode($modules));
+            file_put_contents($modulesPath . '/' . $prestashopVersion . '.json', json_encode($modules));
```

Comment:
File system issue to test here as well ?
```suggestion
            if (false === file_put_contents($modulesPath . '/' . $prestashopVersion . '.json', json_encode($modules))) {
                throw new WhateverException(sprintf('Failed to write file "%s"', $modulesPath . '/' . $prestashopVersion));
            }
```